crates/astria-sequencer/src/transaction/state_ext.rs

```diff
@@ -26,19 +36,35 @@ impl From<&SignedTransaction> for TransactionContext {
     fn from(value: &SignedTransaction) -> Self {
         Self {
             address_bytes: value.address_bytes(),
+            transaction_id: value.id(),
+            position_in_source_transaction: Some(0),
         }
     }
 }
 
 pub(crate) trait StateWriteExt: StateWrite {
-    fn put_current_source(&mut self, transaction: impl Into<TransactionContext>) {
+    fn put_transaction_context(
+        &mut self,
+        transaction: impl Into<TransactionContext>,
+    ) -> TransactionContext {
         let context: TransactionContext = transaction.into();
         self.object_put(current_source(), context);
+        context
     }
 
     fn delete_current_source(&mut self) {
         self.object_delete(current_source());
     }
+
+    #[instrument(skip_all)]
+    fn set_position_in_source_transaction(&mut self, val: u64) -> Result<()> {
+        let mut context = self
+            .get_current_source()
+            .context("failed to get current source")?;
+        context.position_in_source_transaction = Some(val);
+        self.object_put(current_source(), context);
+        Ok(())
+    }
 }
 
 pub(crate) trait StateReadExt: StateRead {
```

Comment:
Commenting on the line below: rename `get_current_source` -> `get_transaction_context`